### PR TITLE
Expose SyncConfiguration.TrustedCAPath

### LINKED ISSUE #1422: Add support for specifying trusted root CA
See the Java PR: https://github.com/realm/realm-java/pull/4591

### PR DESCRIPTION
**Pull request by nirinchev**
## Description

Fixes #1422 

Also fixes a case when `client_validate_ssl` wasn't respected when opening a realm with `GetInstanceAsync`.

##  TODO

* [x] Changelog entry
* [x] Tests (if applicable)
* [x] Update PCL (if applicable)

### CODE REVIEW

File: Realm/Realm.Sync/SyncConfiguration.cs [hunk 1]
```diff
@@ -87,11 +109,18 @@ internal override Realm CreateRealm(RealmSchema schema)
 
         private Native.SyncConfiguration ToNative()
         {
+            if (!string.IsNullOrEmpty(TrustedCAPath) &&
+                !File.Exists(TrustedCAPath))
+            {
+                throw new RealmFileNotFoundException($"File not found: {TrustedCAPath}");
```
Comment (kristiandupont) [inline Realm/Realm.Sync/SyncConfiguration.cs]: Is that the proper exception here? (In fact, I am not even sure why we have an exception by that name?)

Comment (nirinchev) [inline Realm/Realm.Sync/SyncConfiguration.cs]: Well... the file's not there, the exception contains "FileNotFound", so I went with it :P Perhaps we can throw the system's `FileNotFoundException`?

Comment (kristiandupont) [inline Realm/Realm.Sync/SyncConfiguration.cs]: Right. Well, I can't think of any reason not to use the system one, unless we for some reason want all exceptions thrown by our SDK to be `Realm*`-exceptions. But that just seems silly to me. As does prefixing them with `Realm` in general, unless they are related to something in a realm, not just in the SDK.

Comment (nirinchev) [inline Realm/Realm.Sync/SyncConfiguration.cs]: I agree - we throw a bunch of system exceptions anyway. I'll update it here, but will leave the rest of the codebase for another time ;)

**Pull request merged**